From 2ed3d951da2ebb10be22df70a58b23ab560a379f Mon Sep 17 00:00:00 2001
From: Den <motometalor@gmail.com>
Date: Tue, 6 Aug 2019 12:17:20 +0300
Subject: [PATCH 1/2] Devices: fsl: mf0300_6dq: use own mf0300
 BoardConfigCommon.mk

mf0300_6qd/BoardConfigCommon.mk is complitely copied from imx6/BoardConfigCommon.mk
---
 mf0300_6dq/BoardConfig.mk       |  2 +-
 mf0300_6dq/BoardConfigCommon.mk | 76 +++++++++++++++++++++++++++++++++
 2 files changed, 77 insertions(+), 1 deletion(-)
 create mode 100644 mf0300_6dq/BoardConfigCommon.mk

diff --git a/mf0300_6dq/BoardConfig.mk b/mf0300_6dq/BoardConfig.mk
index f8e39f6..ce5a0d0 100644
--- a/mf0300_6dq/BoardConfig.mk
+++ b/mf0300_6dq/BoardConfig.mk
@@ -3,7 +3,7 @@
 #
 include device/fsl/imx6/soc/imx6dq.mk
 include device/fsl/mf0300_6dq/build_id.mk
-include device/fsl/imx6/BoardConfigCommon.mk
+include device/fsl/mf0300_6dq/BoardConfigCommon.mk
 ifeq ($(PREBUILT_FSL_IMX_CODEC),true)
 -include $(FSL_CODEC_PATH)/fsl-codec/fsl-codec.mk
 endif
diff --git a/mf0300_6dq/BoardConfigCommon.mk b/mf0300_6dq/BoardConfigCommon.mk
new file mode 100644
index 0000000..63e9b89
--- /dev/null
+++ b/mf0300_6dq/BoardConfigCommon.mk
@@ -0,0 +1,76 @@
+#
+# Product-specific compile-time definitions.
+#
+
+TARGET_BOARD_PLATFORM := imx6
+TARGET_CPU_ABI := armeabi-v7a
+TARGET_CPU_ABI2 := armeabi
+TARGET_ARCH := arm
+TARGET_ARCH_VARIANT := armv7-a-neon
+TARGET_CPU_VARIANT := cortex-a9
+ARCH_ARM_HAVE_TLS_REGISTER := true
+
+TARGET_NO_BOOTLOADER := true
+TARGET_NO_KERNEL := false
+TARGET_NO_RECOVERY := false
+TARGET_NO_RADIOIMAGE := true
+
+BOARD_SOC_CLASS := IMX6
+
+#BOARD_USES_GENERIC_AUDIO := true
+BOARD_USES_ALSA_AUDIO := true
+BOARD_HAVE_BLUETOOTH := true
+USE_CAMERA_STUB := false
+
+BOARD_HAVE_WIFI := true
+
+BOARD_NOT_HAVE_MODEM := false
+BOARD_MODEM_VENDOR := HUAWEI
+BOARD_MODEM_ID := EM750M
+BOARD_MODEM_HAVE_DATA_DEVICE := true
+BOARD_HAVE_IMX_CAMERA := true
+BOARD_HAVE_USB_CAMERA := false
+
+TARGET_USERIMAGES_BLOCKS := 204800
+
+# Enable dex-preoptimization to speed up first boot sequence
+ifeq ($(HOST_OS),linux)
+   ifeq ($(TARGET_BUILD_VARIANT),user)
+	ifeq ($(WITH_DEXPREOPT),)
+	    WITH_DEXPREOPT := true
+	endif
+   endif
+endif
+# for ums config, only export one partion instead of the whole disk
+UMS_ONEPARTITION_PER_DISK := true
+
+PREBUILT_FSL_IMX_CODEC := true
+PREBUILT_FSL_IMX_OMX := false
+PREBUILT_FSL_IMX_GPU := true
+
+# override some prebuilt setting if DISABLE_FSL_PREBUILT is define
+ifeq ($(DISABLE_FSL_PREBUILT),GPU)
+PREBUILT_FSL_IMX_GPU := false
+else ifeq ($(DISABLE_FSL_PREBUILT),ALL)
+PREBUILT_FSL_IMX_GPU := false
+endif
+
+# for kernel/user space split
+# comment out for 1g/3g space split
+# TARGET_KERNEL_2G := true
+
+BOARD_BOOTIMAGE_PARTITION_SIZE := 33554432
+BOARD_RECOVERYIMAGE_PARTITION_SIZE := 33554432
+
+BOARD_SYSTEMIMAGE_PARTITION_SIZE := 1610612736
+BOARD_CACHEIMAGE_PARTITION_SIZE := 444596224
+BOARD_VENDORIMAGE_PARTITION_SIZE := 117440512
+BOARD_VENDORIMAGE_FILE_SYSTEM_TYPE = ext4
+TARGET_COPY_OUT_VENDOR := vendor
+
+BOARD_FLASH_BLOCK_SIZE := 4096
+TARGET_RECOVERY_UI_LIB := librecovery_ui_imx
+
+-include device/google/gapps/gapps_config.mk
+-include $(FSL_RESTRICTED_CODEC_PATH)/fsl-restricted-codec/fsl_ms_codec/BoardConfig.mk
+-include $(FSL_RESTRICTED_CODEC_PATH)/fsl-restricted-codec/fsl_real_dec/BoardConfig.mk
-- 
2.17.1

